### PR TITLE
Stop erroneous propagation of MapToLocal changes (Lidarr#2443)

### LINKED ISSUE #2443: [BUG]: On Release Import trigger not working for Discord Connection
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When a release has finished downloading, and Lidarr has imported it, _and_ you've set up a Connection for Discord, then a message is not being sent to Discord to confirm import of the release.

### Expected Behavior

When a release has been imported, 'On Release Import' trigger should activate and send a message to Discord to confirm the import

### Steps To Reproduce

1. Set up a Discord Connection in Lidarr
2. Grab a release
3. Lidarr will send a message to Discord to say a release has been Grabbed
4. Wait for Lidarr to import the release
5. There will be no message to confirm the release import in Discord, even though the trigger is enabled

### Environment

```markdown
- OS: Unraid OS 6.9.2
- Lidarr: 0.8.1.2135
- Docker Install: Yes
- Using Reverse Proxy: No
```

### What branch are you running?

Master

### Anything else?

trace logs excerpt of error:  [https://hastebin.com/vekisijoqi.yaml](https://hastebin.com/vekisijoqi.yaml)

Other triggers are working (e.g. On Grab)

[AB#1750](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/1750)

### PR DESCRIPTION
**Pull request by ac3d912**
#### Database Migration
NO

#### Description
tl;dr: Prevent changing the url for MediaCover images to a local resource when processing album change notifications.

The web API is making changes to the items in the Images list of the Album resource. Since it is using a direct reference to `model.Images` the `MediaCover` items in the list are being modified to local resources by `MapCoversToLocal(resource);`. This causes problems when the source of the model is a notification that is being sent to many consumers.

I originally put this in `AlbumResourceMapper->ToResource<AlbumResource>`; however, that gets called every time a model is converted to a resource (all web grabs) which is overkill. Putting it here limits unnecessary clones.

To ease cloning a MediaCover item, I added ICloneable, since it's a structure consisting of only primitive types.

#### Screenshot (if UI related)
N/A

#### Todos
- [ ] Tests
- [x] [Wiki Updates](https://wiki.servarr.com) - N/A

#### Issues Fixed or Closed by this PR

* Fixes #2443

### CODE REVIEW

**Comment by ac3d912** [comment]
Closing w/out merging. Looks like it was fixed with https://github.com/Lidarr/Lidarr/pull/2764 specifically: https://github.com/Lidarr/Lidarr/pull/2764/commits/120a7477721959b70c187abf1b4d82dbfa2cb9bf

Verified today that Discord notifications are working.
